ledger/store/trackerdb/testsuite/onlineaccounts_kv_test.go

```diff
@@ -0,0 +1,554 @@
+			MicroAlgos:     microAlgos,
+			RewardsBase:    rewardBase,
+		}
+		normalizedBal := data.NormalizedOnlineBalance(t.proto)
+
+		// write
+		_, err := oaw.InsertOnlineAccount(addr, normalizedBal, data, updRound, lastValid)
+		require.NoError(t, err)
+
+		testData = append(testData, addr)
+	}
+
+	// read (all)
+	poA, err := ar.AccountsOnlineTop(basics.Round(0), 0, 10, t.proto)
+	require.NoError(t, err)
+	require.Contains(t, poA, testData[9]) // most money
+	require.Contains(t, poA, testData[0]) // least money
+
+	// read (just a few)
+	poA, err = ar.AccountsOnlineTop(basics.Round(0), 1, 2, t.proto)
+	require.NoError(t, err)
+	require.Len(t, poA, 2)
+	require.Contains(t, poA, testData[8]) // (second most money, we skipped 1)
+	require.Contains(t, poA, testData[7]) // (third, we only have 2 items)
+}
+
+func CustomTestLookupOnlineAccountDataByAddress(t *customT) {
+	oaw, err := t.db.MakeOnlineAccountsOptimizedWriter(true)
+	require.NoError(t, err)
+
+	ar, err := t.db.MakeAccountsReader()
+	require.NoError(t, err)
+
+	// generate some test data
+	addrA := RandomAddress()
+	updRoundA := uint64(400)
+	lastValidA := uint64(500)
+	dataA := trackerdb.BaseOnlineAccountData{
+		BaseVotingData: trackerdb.BaseVotingData{},
+		MicroAlgos:     basics.MicroAlgos{Raw: uint64(100)},
+		RewardsBase:    uint64(200),
+	}
+	normalizedBalA := dataA.NormalizedOnlineBalance(t.proto)
+
+	refA, err := oaw.InsertOnlineAccount(addrA, normalizedBalA, dataA, updRoundA, lastValidA)
+	require.NoError(t, err)
+
+	//
+	// test
+	//
+
+	// check non-existing account
+	nonExistingAddr := RandomAddress()
+	_, _, err = ar.LookupOnlineAccountDataByAddress(nonExistingAddr)
+	require.Error(t, err)
+	require.Equal(t, trackerdb.ErrNotFound, err) // check the error type
+
+	// read existing addr
+	readRef, readData, err := ar.LookupOnlineAccountDataByAddress(addrA)
+	require.NoError(t, err)
+	require.Equal(t, refA, readRef) // check ref is the same
+	// the method returns raw bytes, parse them
+	var badA trackerdb.BaseOnlineAccountData
+	err = protocol.Decode(readData, &badA)
+	require.NoError(t, err)
+	require.Equal(t, dataA, badA)
+}
+
+func CustomTestOnlineAccountTotals(t *customT) {
+	aw, err := t.db.MakeAccountsWriter()
+	require.NoError(t, err)
+
+	oaor, err := t.db.MakeOnlineAccountsOptimizedReader()
+	require.NoError(t, err)
+
+	// generate some test data
+	roundParams := []ledgercore.OnlineRoundParamsData{
+		{OnlineSupply: 100},
+		{OnlineSupply: 42},
+		{OnlineSupply: 9000},
+	}
+	err = aw.AccountsPutOnlineRoundParams(roundParams, basics.Round(3))
+	require.NoError(t, err)
+
+	//
+	// test
+	//
+
+	// lookup totals
+	totals, err := oaor.LookupOnlineTotalsHistory(basics.Round(4))
+	require.NoError(t, err)
+	require.Equal(t, basics.MicroAlgos{Raw: uint64(42)}, totals)
+
+	// lookup not found
+	_, err = oaor.LookupOnlineTotalsHistory(basics.Round(121))
+	require.Error(t, err)
+	require.Equal(t, trackerdb.ErrNotFound, err)
+}
+
+func CustomTestOnlineAccountsDelete(t *customT) {
```

Comment:
there is a bigger test `TestOnlineAccountsDeletion` and I do think the new implementation have to at least pass it but ideally extend it to 3+ accounts and more rounds.